pkg/controllers/util/federatedclient/podinformer_test.go

```diff
@@ -0,0 +1,346 @@
+	"k8s.io/client-go/kubernetes/fake"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	fakecorev1 "k8s.io/client-go/kubernetes/typed/core/v1/fake"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/kubewharf/kubeadmiral/pkg/controllers/util"
+)
+
+var errLimitExceeded = errors.New("limit exceeded")
+
+type admiralFakePods struct {
+	corev1client.PodInterface
+	concurrentLists *atomic.Int64
+}
+
+func (c *admiralFakePods) List(ctx context.Context, opts metav1.ListOptions) (result *corev1.PodList, err error) {
+	cur := c.concurrentLists.Add(-1)
+	defer c.concurrentLists.Add(1)
+	if cur < 0 {
+		panic(errLimitExceeded)
+	}
+	return c.PodInterface.List(ctx, opts)
+}
+
+type admiralFakeCoreV1 struct {
+	fakecorev1.FakeCoreV1
+	concurrentLists *atomic.Int64
+}
+
+func (c *admiralFakeCoreV1) Pods(namespace string) corev1client.PodInterface {
+	return &admiralFakePods{c.FakeCoreV1.Pods(namespace), c.concurrentLists}
+}
+
+var _ kubeclient.Interface = &admiralFakeClientset{}
+
+type admiralFakeClientset struct {
+	*fake.Clientset
+	concurrentLists *atomic.Int64
+}
+
+func (c *admiralFakeClientset) CoreV1() corev1client.CoreV1Interface {
+	return &admiralFakeCoreV1{fakecorev1.FakeCoreV1{Fake: &c.Clientset.Fake}, c.concurrentLists}
+}
+
+func newAdmiralFakeClientset(concurrentLists int64, objects ...runtime.Object) *admiralFakeClientset {
+	client := fake.NewSimpleClientset(objects...)
+	var listers atomic.Int64
+	listers.Add(concurrentLists)
+	return &admiralFakeClientset{
+		Clientset:       client,
+		concurrentLists: &listers,
+	}
+}
+
+func setupFakeClient(concurrentLists int64, numPods int) (pods []runtime.Object, client kubeclient.Interface) {
+	pods = make([]runtime.Object, numPods)
+	for i := 0; i < numPods; i++ {
+		ns := fmt.Sprintf("test-ns-%d", i%10)
+		name := fmt.Sprintf("test-name-%d", i)
+		pods[i] = newPod(ns, name)
+	}
+	return pods, newAdmiralFakeClientset(concurrentLists, pods...)
+}
+
+func newPod(ns, name string) *corev1.Pod {
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Labels:    map[string]string{"foo": "bar"},
+			Namespace: ns,
+		},
+		Spec: corev1.PodSpec{
+			Overhead: corev1.ResourceList{
+				corev1.ResourceCPU: resource.MustParse("1m"),
+			},
+			InitContainers: []corev1.Container{
+				{
+					Name: "initcontainer",
+					Resources: corev1.ResourceRequirements{
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU: resource.MustParse("1m"),
+						},
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU: resource.MustParse("1m"),
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name: "container",
+					Resources: corev1.ResourceRequirements{
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU: resource.MustParse("10m"),
+						},
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU: resource.MustParse("10m"),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+//nolint:gocyclo
```

Comment:
What was the complexity gocyclo reported? Any chance we can address the lint instead of suppressing it?